### PR TITLE
Fixes crafting multiples of recipes in a sublist

### LINKED ISSUE #26273: Bamboo tatami crafting bug
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1641

### Issue Description

When trying to craft 20x bamboo tatami mats at once, instead of doing that it starts crafting bamboo punji sticks instead
image_2024-07-22_202924366
image_2024-07-22_202934964

### What did you expect to happen?

I get 20x bamboo tatami mats

### What happened instead?

i got punji sticks...

### Why is this bad/What are the consequences?

misleading and annoying

### Steps to reproduce the issue.

Try to craft 20x tatami mats

### When did the problem start happening?

never crafted it before until today

### Extra information

nope

### Relevant log output/runtime error

```DM
nope
```

### PR DESCRIPTION
**Pull request by Migratingcocofruit**
## What Does This PR Do
Fixes #26273
Adds sublist identifier to the crafting buttons of items in sublists so they don't craft things from the main list.

## Why It's Good For The Game
It Fixes it

## Testing
Made 20 tatami mats of each colour.

## Changelog
:cl:
fix: Fixes crafting multiples of recipes in a sublist
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -180,9 +180,9 @@
 				var/list/multipliers = list(5, 10, 25)
 				for(var/n in multipliers)
 					if(max_multiplier >= n)
-						t1 += " <A href='byond://?src=[UID()];make=[i];multiplier=[n]'>[n * R.res_amount]x</A>"
+						t1 += " <A href='byond://?src=[UID()];sublist=[recipes_sublist];make=[i];multiplier=[n]'>[n * R.res_amount]x</A>"
```
Comment (Burzah) [inline code/game/objects/items/stacks/stack.dm:183]:
```suggestion
						t1 += " <a href='byond://?src=[UID()];sublist=[recipes_sublist];make=[i];multiplier=[n]'>[n * R.res_amount]x</a>"
```

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -180,9 +180,9 @@
...
 				if(!(max_multiplier in multipliers))
-					t1 += " <A href='byond://?src=[UID()];make=[i];multiplier=[max_multiplier]'>[max_multiplier * R.res_amount]x</A>"
+					t1 += " <A href='byond://?src=[UID()];sublist=[recipes_sublist];make=[i];multiplier=[max_multiplier]'>[max_multiplier * R.res_amount]x</A>"
```
Comment (Burzah) [inline code/game/objects/items/stacks/stack.dm:185]:
```suggestion
					t1 += " <a href='byond://?src=[UID()];sublist=[recipes_sublist];make=[i];multiplier=[max_multiplier]'>[max_multiplier * R.res_amount]x</a>"
```

Comment (Migratingcocofruit) [reply code/game/objects/items/stacks/stack.dm]: done

File: code/game/objects/items/stacks/stack.dm (hunk 1 above)
Comment (Migratingcocofruit) [reply code/game/objects/items/stacks/stack.dm]: done

**Review by Sirryan2002 — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]

**Pull request merged**